explainability-service/src/main/java/org/kie/trustyai/service/endpoints/metrics/BaseEndpoint.java

```diff
@@ -66,9 +66,8 @@ public String getMetricName() {
     public Response deleteRequest(ScheduleId request) {
 
         final UUID id = request.requestId;
-
         if (scheduler.getRequests(this.name).containsKey(id)) {
-            scheduler.getRequests(this.name).remove(request.requestId);
```

Comment:
I don't know if this pattern is used for other use cases too (i.e. .add()) but overall we should prevent this code to be accessible (aka `getMap(...).remove(...)`) because it is not safe.
To make sure this is enforced we can wrap the value in `getRequests(..)` with [`Collections.unmodifiableMap(...)`](https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html#unmodifiableMap-java.util.Map-) so that it fails is the code tries to modify the object. Unfortunately Java has no "real" immutable collections so we will need to check all the usages and maybe introduce some more methods like add/update or similar to `scheduler`.

I'm fine if we postpone this refactor to a future ticket to keep this limited to the remove scenario. Up to you

Reply:
done